src/Platform/Microsoft.Testing.Platform/IPC/Serializers/CommandLineOptionMessagesSerializer.cs

```diff
@@ -0,0 +1,157 @@
+
+            switch (fieldId)
+            {
+                case CommandLineOptionMessagesFields.ModuleName:
+                    moduleName = ReadString(stream);
+                    break;
+
+                case CommandLineOptionMessagesFields.CommandLineOptionMessageList:
+                    commandLineOptionMessages = ReadCommandLineOptionMessagesPayload(stream);
+                    break;
+
+                default:
+                    // If we don't recognize the field id, skip the payload corresponding to that field
+                    SetPosition(stream, stream.Position + fieldSize);
+                    break;
+            }
+        }
+
+        return new CommandLineOptionMessages(moduleName, commandLineOptionMessages is null ? [] : [.. commandLineOptionMessages]);
+    }
+
+    private static List<CommandLineOptionMessage> ReadCommandLineOptionMessagesPayload(Stream stream)
+    {
+        List<CommandLineOptionMessage> commandLineOptionMessages = [];
+
+        int length = ReadInt(stream);
+        for (int i = 0; i < length; i++)
+        {
+            string name = string.Empty, description = string.Empty, arity = string.Empty;
+            bool isHidden = false, isBuiltIn = false;
+
+            int fieldCount = ReadShort(stream);
+
+            for (int j = 0; j < fieldCount; j++)
+            {
+                int fieldId = ReadShort(stream);
+                var fieldSize = ReadInt(stream);
+
+                switch (fieldId)
+                {
+                    case CommandLineOptionMessageFields.Name:
+                        name = ReadString(stream);
+                        break;
+
+                    case CommandLineOptionMessageFields.Description:
+                        description = ReadString(stream);
+                        break;
+
+                    case CommandLineOptionMessageFields.Arity:
+                        arity = ReadString(stream);
+                        break;
+
+                    case CommandLineOptionMessageFields.IsHidden:
+                        isHidden = ReadBool(stream);
+                        break;
+
+                    case CommandLineOptionMessageFields.IsBuiltIn:
+                        isBuiltIn = ReadBool(stream);
+                        break;
+
+                    default:
+                        SetPosition(stream, stream.Position + fieldSize);
+                        break;
+                }
+            }
+
+            commandLineOptionMessages.Add(new CommandLineOptionMessage(name, description, arity, isHidden, isBuiltIn));
+        }
+
+        return commandLineOptionMessages;
+    }
+
+    public void Serialize(object objectToSerialize, Stream stream)
+    {
+        RoslynDebug.Assert(stream.CanSeek, "We expect a seekable stream.");
+
+        CommandLineOptionMessages commandLineOptionMessages = (CommandLineOptionMessages)objectToSerialize;
+
+        WriteShort(stream, (short)GetFieldCount(commandLineOptionMessages));
```

Comment:
"short" it in the GetFieldCount is we expect that, also we should use an `ushort` that's wider.